eth2/utils/tree_hash/src/merkle_hasher.rs

```diff
@@ -274,22 +274,20 @@ impl MerkleHasher {
         loop {
             if let Some(root) = self.root {
                 break Ok(root);
+            } else if let Some(node) = self.half_nodes.last() {
+                let right_child = node.id * 2 + 1;
```

Comment:
`mul` and `add` can be checked here since we return `Result`

Reply:
Oh yeah, we could. This refactor was just leftover from me trying to fix generic clippy lints project-wide (which I ended up disabling for now), and this crate isn't being linted currently. I agree we could expand the usage of checked arithmetic though, so I've made an issue to track that: https://github.com/sigp/lighthouse/issues/1013